### ISSUE TITLE
Show coastal defence for all, show barracks/port/airport for player and allies

### ISSUE THREAD

**Issue by hugomflavio** [longturn/freeciv21#787]
Make the client able to interpret the presence of coastal defence and render a different city graphic. The graphic would have to change with the presence of walls too (e.g. cannons on the floor vs cannons on top of the wall). Simple mock-up of the coastal defence below. It would also make sense to be able to show the SDI defences. The visibility of each defensive element (walls, coastal defence and sdi) should be ruleset dependent.

![image](https://user-images.githubusercontent.com/40389021/147391700-cd1aa6eb-ece6-401c-9006-3e9fdb3b45e3.png)

Additionally, allow a visual for barracks, ports and airports (airports already exists, I think?). This could be made 1) visible for all, 2) visible for the player and allies, 3) visible for the player only, or 3) not visible at all depending on the ruleset. Simple mock-up:

![image](https://user-images.githubusercontent.com/40389021/147391841-e6691966-15f9-45a5-b576-1bc6ae007271.png)

The style of the drawings could be adapted for each tileset once the mechanisms are present.

**Comment by lmoureaux** [comment]
There can be up to 7 (`NUM_WALL_TYPES`, used in network protocol) sprites in addition to the walls. They have formats like `city.modern_bldg_$i_0`, with `$i` from 0 to 6. The picked sprite comes from the value of the `Visible_Wall` effect in the ruleset.

With a default ruleset, one can probably get different sprites when there are walls in addition to the Great Wall ($i=1 if not, 2 if yes).

**Comment by hugomflavio** [comment]
Does that mean we already have what we need to be able to get different sprites going for walls/coastal/SDI? Sorry, I got a bit lost on the technical wording :sweat_smile:

**Comment by lmoureaux** [comment]
Yes, if the ruleset author does it right :tm:. But the way it works is inconsistent, since there's also a building flag called `VisibleByOthers` that does the very same thing...
